### ISSUE TITLE
Reasons for assert torch.all(tensor >= 0)

### ISSUE THREAD

**Issue by PavelOstyakov** [photosynthesis-team/piq#168]
Inside the _validate_input function, there is a check for values of the input tensor to be >=0: https://github.com/photosynthesis-team/piq/blob/master/piq/utils/common.py#L46

However, this introduces difficulties when using some of the losses from piq. For example, I would like to use DISTS as a loss function for training a neural network without an activation function at the end. 

Could you please clarify the reasons why this check is required?

**Comment by zakajd** [comment]
Hi
Thanks for noticing this, that’s actually a bug. Check for non-negative values is important for many other metrics that expect images as inputs with range [0, ‘data_range’] to avoid, for example, calculation of SSIM on negative values which doesn’t make sense.

Will add a hotfix tomorrow

**Comment by franec94** [comment]
I also have a comment for the same issue stated by PavelOstyakov right at the very beginning of this issue, that is:

- Inside the _validate_input function, there is a check for values of the input tensor to be >=0: https://github.com/photosynthesis-team/piq/blob/master/piq/utils/common.py#L46

I want ask the following, does **PSNR** all values from pytorch tensors, to be compared, need  to be positive in order to be calculated ? Because, in its own derivation, PSNR makes use of **MSE,** so that it is relevant the sign that pytorch tensor components carry on even if the pytorch tensor are expressed at least as **size = {batch_size x C x H x W }**.

Since I've experienced some troubles when using **piq.psnr** module with output from a Siren-Based Deep Learning Architecture that processed grayscale images and the ouptut I obtained did contain negative values, while compressing the input image.

**Comment by zakajd** [comment]
@franec94 thats a good question.
As far as I understand, PSNR should be computed between positive images, because we also need value range for computation. See [Wikipedia](https://en.wikipedia.org/wiki/Peak_signal-to-noise_ratio) formula.

For example, if you try to compute PSNR between 2 tensors with values in [-127, 128], your maximum value is 128, but actual range is 255. In this case you, theoretically, can compute PSNR between those tensors by setting `data_range = 255`.  But when you deal with raw logits from neural network it's not possible, because you never know the range of output values. Is it [-5, 5], [-10, 10] or [-1, 1]? 
I would recommend to clip values into [0, 1] by calling `.sigmoid()` method on both tensors before computation.

**Comment by franec94** [comment]
@zakajd  thank you for your answer as well as your proposal for `.sigmoid(). I will also check again  [Wikipedia](https://en.wikipedia.org/wiki/Peak_signal-to-noise_ratio) formula, for better understanding my faults, whiel coding a Siren Arch for compression purposes.

however, if I will still have some doubts, I desire to reask something to you.
